packages/expo-updates/ios/EXUpdates/AppLoader/EXUpdatesAppLoader.m

```diff
@@ -59,6 +59,29 @@ - (void)_reset
   _errorBlock = nil;
 }
 
+# pragma mark - class methods
+- (NSArray<NSUUID *> *)storedUpdateIds:(EXUpdatesAppLoaderErrorBlock)errorBlock
```

Comment:
I am open to being convinced, but as I noted on the initial commit, I don't think it makes sense for this method to live in the AppLoader class (which has a specific purpose -- take a URL that points to an update manifest, and download all the assets that comprise that update).

I think the logic here, which is really just mapping an array of updates to IDs, could just live inside of the UpdatesDevLauncherController (or potentially UpdatesDatabase).

Reply:
Let's discuss this tomorrow. I did it this way because the AppLoader already has a reference to the DB and has methods that call into it.

Reply:
Gotcha. Yeah definitely down to discuss, but since I'm ooo tomorrow just posting here too so I'm not blocking you! -- I would suggest trying to grab the database reference from the EXUpdatesAppController singleton (which owns it) after initializing it. I *think* it should be simpler that way.

And you're right, my apologies -- I was oversimplifying above. `EXUpdatesRemoteAppLoader` loads an update from a URL; `EXUpdatesEmbeddedAppLoader` loads an "update" that is embedded in the .ipa/.apk (we try to treat these embedded bundles as similarly as possible to OTA updates, so even though they're already on disk, we still load them into SQLite and copy all the assets to the same location), and `EXUpdatesAppLoader` is a superclass that holds the shared logic between these two loader classes.

Reply:
Ah thanks for explaining in more detail the design of these classes. I'm wondering if what we should do instead is put this in a new subclass of `EXUpdatesAppLoader`, like `EXupdatesOfflineAppLoader`, that would eventually be responsible for loading an update and its assets from SQLite.

Reply:
@esamelson I reworked this to implement the new method directly in `EXUpdatesDevLauncherController` as you suggested, grabbing the database reference from the `EXUpdatesAppController` shared instance.  That compiles, but seems to be very difficult to set up unit tests for.  Some of the initialization code for the singletons will not work in the unit test environment.

Let's meet on Thursday and discuss the issues I ran into. At this point I'd prefer to either leave this change as is, or maybe create a new `EXUpdatesOfflineAppLoader` for offline update loading.  In the meantime I've started work on doing this in Android.